Porting to vegman

nest1ing opened the pull request:
This patch set fixes `yadro-snmp` build for Vegman and makes SNMP tables work properly.
But a couple of issues still present on Vegman, that will be solved with the further path sets.

nest1ing committed:
agent: dbus: fix sensor value type

Sensors value type was changed from `int64_t` to `double` by upstream.

This commit fixes the received data handler.
For the backward compatible the sensor value representation in the
MIB-file and in the replies kept as it was before.

Signed-off-by: Alexander Filippov <a.filippov@yadro.com>

nest1ing committed:
agent: dbus: filter sensor objects

VEGMAN has some extra D-Bus objects in `/xyz/openbmc_project/sensors`,
that are not sensors in fact.
This commit adds a filter to skipping such objects.

Signed-off-by: Alexander Filippov <a.filippov@yadro.com>

nest1ing committed:
agent: sdeventplus: fix signal handler

`sdevent/event.hpp` was written as a copy of usable code from the early
and poor version of `sdeventplus` library.

`sdeventplus` now is stable and already present in the BMC firmware.
It also has a lot of features that are usable for this project.

This commit replaces our poor version with external full version and
fixes the crash on application termination by UNIX signal.

Signed-off-by: Alexander Filippov <a.filippov@yadro.com>

nest1ing committed:
agent: snmp: check master connection

This commit adds a timer that calls the check the state of the
connection to the master daemon and a try to reconnect when it was lost.

Signed-off-by: Alexander Filippov <a.filippov@yadro.com>

nest1ing committed:
agent: Make debug output optional

This commit replaces `TRACE_DEBUG` macros with `DEBUGMSGTL` that is
hidden output by default and may be enabled by `-D` command line
argument.

Signed-off-by: Alexander Filippov <a.filippov@yadro.com>

nest1ing committed:
agent: Remove unused build-time param

This commit removes `host instance` build time parameter as it is excess
for this project.

Signed-off-by: Alexander Filippov <a.filippov@yadro.com>

alatarum commented on agent/yadro/sensors.cpp:
```diff
@@ -93,15 +91,14 @@ struct Sensor
-                                                   0, false, // CriticalLow
-                                                   0, false, // CriticalHigh
-                                                   1)        // Scale
+        phosphor::snmp::data::table::Item<double, double, bool, double, bool,
```
Don't we need to support old platforms? How does it work now?

AlexanderAmelkin replied:
I guess old platforms use old version of this repo. And if we decide to upgrade them, we'll upgrade the interface as well, just as we do when merging updates from upstream for other projects.

nest1ing replied:
Yes, exactly so.

artemsen commented on configure.ac:
```diff
@@ -23,6 +23,11 @@ PKG_CHECK_MODULES([SDBUSPLUS], [sdbusplus],,\
 AC_SUBST(SDBUSPLUS_LIBS)
 
+PKG_CHECK_MODULES([SDEVENTPLUS], [sdeventplus],,\
+    AC_MSG_ERROR(["Required sdeventplus package."]))
```
Antipattern: https://github.com/openbmc/docs/blob/master/anti-patterns.md#explicit-ac_msg_error-on-pkg_check_modules-failure

nest1ing replied:
fixed

AlexanderAmelkin commented on agent/main.cpp:
```diff
@@ -101,45 +103,49 @@ int parse_args(int argc, char** argv)
+    if (sigemptyset(&ss) < 0 || sigaddset(&ss, SIGTERM) < 0 ||
+        sigaddset(&ss, SIGINT) < 0)
+    {
+        TRACE_ERROR("Failed to setup signals hanlders.\n");
```
signals -> signal

nest1ing replied:
fixed

AlexanderAmelkin commented on agent/main.cpp:
```diff
@@ -101,45 +103,49 @@ int parse_args(int argc, char** argv)
     return rc;
 }
 
-void setup_signals(sdevent::event::Event& evt)
```
Why did you decide to move it from a helper function into main() ?

nest1ing replied:
There are two objects (for each signal) that should be alive during main loop working.
And this function is part of their initialization and called only once.

AlexanderAmelkin commented on agent/yadro/sensors.cpp:
```diff
@@ -131,14 +128,11 @@ struct Sensor
+            // Required for TEXTUAL-CONVENTION in YADRO-MIB.txt
             switch (folder[n + 1])
             {
-                case ST_TEMPERATURE:
-                case ST_VOLTAGE:
-                case ST_CURRENT:
-                case ST_POWER:
```
What happened to these values? Why are they not needed anymore?

nest1ing replied:
The logic was changed. I would say that the logic was inverted. Now only `ST_TACHOMETER` requires the correction.

nest1ing committed:
agent: remove sdbusplus::message::variant

`sdbusplus::message::variant` was completely removed in the latest
version of `sdbusplus` library. This commit removes it from our code
too.

Signed-off-by: Alexander Filippov <a.filippov@yadro.com>
